main.go

```diff
@@ -1,343 +1,103 @@
-//certlint contains functions to check certificates for standards compliance.
 package main
 
 import (
 	"bufio"
+	"encoding/base64"
 	"encoding/json"
-	"errors"
 	"flag"
 	"fmt"
-	"io"
-	"io/ioutil"
+	log "github.com/Sirupsen/logrus"
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/zlint"
 	"os"
+	"runtime"
 	"sync"
-
-	"github.com/zmap/zlint/lints"
-	"github.com/zmap/zlint/zlint"
-	"github.com/zmap/zlint/zlint/ringbuff"
 )
 
-const CHUNKSIZE int = 10000    //number of certs per work unit, must be >=1
-const DEFAULT_THREADS uint = 4 //default number of processing threads for -threads mode, must be >=1
-
 var ( //flags
-	inPath      string
-	outPath     string
-	outStat     string
-	multi       bool
-	threaded    bool
-	numThreads  uint
-	prettyPrint bool
-)
-
-var ( //sync values for -threads
-	inBuffer      ringbuff.RingBuffer
-	outBuffer     ringbuff.RingBuffer
-	poisonBarrier sync.WaitGroup //used prevent outBuffer from being poisoned before Enqueueing is complete
-	//barrier to prevent the early main thread exits
-	exittex       sync.Mutex //guards writeComplete and mainWait
-	writeComplete bool       //output complete if true
-	mainWait      *sync.Cond //condition variable for the main thread, associated with exittex
+	inPath         string
+	outPath        string
+	multi          bool
+	threaded       bool
+	numCertThreads int
+	prettyPrint    bool
+	numProcs       int
+	channelSize    int
 )
 
 func init() {
 	flag.StringVar(&inPath, "in-file", "", "File path for the input certificate(s).")
 	flag.StringVar(&outPath, "out-file", "-", "File path for the output JSON.")
-	flag.StringVar(&outStat, "out-stat", "-", "File path for the output stats.")
 	flag.BoolVar(&prettyPrint, "list-lints-json", false, "Use this flag to print supported lints in JSON format, one per line")
 	flag.BoolVar(&multi, "multi", false, "Use this flag to specify inserting many certs at once. Certs in this mode must be Base64 encoded DER strings, one per line.")
 	flag.BoolVar(&threaded, "threads", false, "Use this flag to specify that -multi mode runs multi-threaded. This has no effect otherwise.")
-	flag.UintVar(&numThreads, "num-threads", DEFAULT_THREADS, "Use this flag to specify the number of threads in -threads mode.  This has no effect otherwise.")
+	flag.IntVar(&numCertThreads, "cert-threads", 1, "Use this flag to specify the number of threads in -threads mode.  This has no effect otherwise.")
+	flag.IntVar(&numProcs, "procs", 1, "Use this flag to specify the number of processes to run on.")
```

Comment:
This should default to 0 or -1. Setting runtime.GOMAXPROCS to 0 or less means "keep it what it was previously".